lib/rubocop/cop/mixin/check_line_breakable.rb

```diff
@@ -222,6 +224,22 @@ def already_on_multiple_lines?(node)
 
         !node.single_line?
       end
+
+      def chained_to_heredoc?(node)
+        # Check if a send
+        receiver = node.receiver
+        return false unless receiver
+
+        while receiver
+          if (receiver.str_type? || receiver.dstr_type? || receiver.xstr_type?) && receiver.heredoc?
+            return true
+          end
+
+          receiver = receiver.receiver
+        end
+
+        false
+      end
```

Comment:
Can you refactor to the following?

```suggestion
      def chained_to_heredoc?(node)
        while (node = node.receiver)
          if (node.str_type? || node.dstr_type? || node.xstr_type?) && node.heredoc?
            return true
          end
        end

        false
      end
```

Reply:
Sure, updated.